server/auth_callout.go

```diff
@@ -118,6 +118,9 @@ func (s *Server) processClientOrLeafCallout(c *client, opts *Options) (authorize
 		}
 		// By default issuer needs to match server config or the requesting account in operator mode.
 		if arc.Issuer != issuer {
+			if arc.IssuerAccount != "" {
+				arc.Issuer = arc.IssuerAccount
```

Comment:
`arc.Issuer` is used in 177 below to see if it is a signing key. This change would seem however to always set the `arc.issuer` to the account root key (`IssuerAccount`).

Since the server is relying on the `aud` of the user claims to indicate the target account, would it be simpler to change 127 to:
```golang
} else if !acc.isAllowedAcount(arc.User.Audience) {
```